### PR TITLE
DOC-1872 Enhance Java data structures guide to emphasize Java Collections Framework implementations

### PR DESCRIPTION
**Pull request by dnault**
Brings the descriptions of the bucket-level methods in line with reality.
Adds a section on Collections Framework integration.

@subalakr  @ingenthr Can you please review for accuracy and style?

### CODE REVIEW

File: content/sdk/java/datastructures.dita [hunk 1]
```diff
@@ -34,23 +34,42 @@
...
-                empty list, set, or queue, use <codeph>bucket.insert(JsonArrayDocument.create("list_id", JsonArray.empty()));</codeph>. To
-                create an empty map, use <codeph>bucket.insert(JsonDocument.create("map_id", JsonObject.empty()));
-</codeph>.</p></section>
-        <section><title>Accessing Data Structures</title>Data structures can be accessed using the
-            appropriate methods. Most data access methods will return a generic type
+        <section><title>Creating a Data Structure</title>
+            <p>
+                Data structures are stored on the Couchbase server as ordinary JSON documents. It is therefore
+                possible to create and reset them using full-document methods.
+                To create a new empty list, set, or queue, use:
```
Comment (subalakr) [inline content/sdk/java/datastructures.dita]: I think it is worth to emphasize on implicit and explicit document creation, might be good to mention that implicit creation is easier using the createDocument flag.

File: content/sdk/java/datastructures.dita [hunk 2]
```diff
@@ -84,7 +103,7 @@ bucket.mapGet("map_id", "name", String.class);  // "mark nunberg"</codeblock>The
                             <row>
                                 <entry><apiname>mapGet</apiname></entry>
                                 <entry>Get an item from a
-                                    map.<codeblock outputclass="language-java">bucket.mapGet("map_id", "some_key", String.class) #=> value</codeblock>If
```
Comment (subalakr) [inline content/sdk/java/datastructures.dita]: Nice find!

File: content/sdk/java/datastructures.dita (hunk 1 above)
Comment (dnault) [inline content/sdk/java/datastructures.dita]: Got it; I'll rearrange so we talk about the document auto-creation with MutationOptionBuilder before introducing explicit creation.

**Comment by subalakr** [comment]
Looks good!

**Comment by amarantha-k** [comment]
@tonyjhillman Subhashni and Matt have reviewed and approved the content. Please complete the final review and merge when done.

**Comment by tonyjhillman** [comment]
Edits Required for the .NET Page
--------------------------------

(http://docs-build.sc.couchbase.com/server/1828/sdk/dotnet/datastructures.html)


In paragraph 4 of "Creating a Data Structure", what is the meaning of the
following?:

	Data structures can be explicitly created and reset using full-document 
	methods, and initialize it with its JSON equivalent.
	
Specifically, what is referred to by "it" and its"? Is this attempting to say "...and
each data structure can be initialized with its JSON equivalent"?



In paragraph one of "Accessing Data Structures", "an ValueResult-like object" should
be "a ValueResult-like object".


In Table 1. Data Structure Operations, in the description for QueuePop, "Remove an item from the end queue" should I think be "Remove an item from the end of the queue".



In the table for CouchbaseSet<TValue>:

In the Description for Contains(TValue), "Checks if an items exists" should be "Check if an item exists".

In the Description for Count, "Returns the total number of items int the set" should be "Returns the total number of items in the set".



In paragraph one of CouchbaseQueue<TValue>, "The CouchbaseQueue<TValue> is implementation" should be "The CouchbaseQueue<TValue> is an implementation".



Under Concurrency, "(Comapre and Swap)" should be "(Compare and Swap)".



Edits Required for the Java Page
--------------------------------

(http://docs-build.sc.couchbase.com/server/1828/sdk/java/datastructures.html)

In paragraph two of Accessing Data Structures, "like if the map key or list index is not found" should be rewritten: though I'm not entirely sure as to what it means. Does it mean "as occurs when the map key or list index is not found"?


In table 1, Data Structure Operations, in the description for mapGet, "an PathNotFoundException" should be 'a PathNotFoundException".


In the description for listGet, ditto.


In the description for QueuePop, "Remove an item from the end queue" should I think
be "Remove an item from the end of the queue".


Under Collections Framework Integration:

The last sentence of paragraph one should be changed to read, "The Map, List, and Queue implementations may contain values of the following types:"

The sentence immediately after the list of types should be changed to read: "The Set implementation may contain values of all the above types except JsonObject and JsonArray."

**Pull request merged**